src/async-split-at.mjs

```diff
@@ -0,0 +1,69 @@
+import ensureAsyncIterable from './internal/ensure-async-iterable'
+import curry from './internal/curry'
+
+function asyncSplitAt (index, iterable) {
+  const iter = ensureAsyncIterable(iterable)[Symbol.asyncIterator]()
+  let position = 0
+  let firstCache = null
+  let firstCacheStart = null
+  let done = false
+
+  async function cleanUp () {
+    if (!done && typeof iter.return === 'function') {
+      await iter.return()
+    }
+  }
+
+  async function * first () {
+    try {
+      while (!done) {
+        if (firstCache) {
+          yield firstCache[position - firstCacheStart]
+        } else {
+          const item = await iter.next()
+          done = item.done
+          if (!done) {
+            yield item.value
+          }
+        }
+        position++
+      }
+    } finally {
+      if (position !== index - 1) {
+        await cleanUp()
+      }
+    }
+  }
+
+  async function * second () {
+    try {
+      if (position < index - 1) {
+        firstCacheStart = position
+        firstCache = new Array(index - position)
+        for (let i = position; i < index; i++) {
+          const item = await iter.next()
```

Comment:
So you should put the promise of the next item into the cache, not the item itself.

Reply:
I could swear I had two comments which explained this, and only one of them showed up.

Reply:
As long as you don't wait on the item promise before the item is in the cache, it will work